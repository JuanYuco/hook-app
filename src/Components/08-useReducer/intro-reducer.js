const initialState = [{
    id: 1,
    todo: 'Comprar Pan',
    donde: false
}];

const todoReducer = ( state = initialState, action ) => {
    if ( action?.type === 'add' ){
        return [ ...state, action.payload ];
    }

    return state;
}

let todos = todoReducer();

const newTodo = {
    id: 2,
    todo: 'Comprar Leche',
    donde: false
}

const addAction = {
    type: 'add',
    payload: newTodo
}

todos = todoReducer( todos, addAction );

console.log(todos);