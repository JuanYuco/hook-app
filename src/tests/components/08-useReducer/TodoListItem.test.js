import { shallow } from "enzyme";
import { TodoListItem } from "../../../Components/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe('Pruebas en <TodoListItem />', () => {
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(
        <TodoListItem 
            todo={ demoTodos[1] }
            index={ 1 }
            handleDelete={ handleDelete }
            handleToggle={ handleToggle }
        />
    );
    test('Debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de llamar la función handleDelete', () => {
        wrapper.find('button') .simulate('click');
        expect( handleDelete ).toHaveBeenCalledWith( demoTodos[1].id );
    });
    
    test('Debe de llamar la función handleToggle', () => {
        wrapper.find('p') .simulate('click');
        expect( handleToggle ).toHaveBeenCalledWith( demoTodos[1].id );
    });
    
    test('Debe de mostrar el texto correctamente', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(`2. ${ demoTodos[1].desc }`)
    });
    
    test('Debe de tener la clase complete si el TODO.done = true', () => {
        let todo = demoTodos[0];
        todo.done = true;
        const wrapper = shallow(
            <TodoListItem 
                todo={ todo }
                index={ 1 }
                handleDelete={ handleDelete }
                handleToggle={ handleToggle }
            />
        );

        expect( wrapper.find('p').hasClass('complete') ).toBe(true);
    });
    
});
