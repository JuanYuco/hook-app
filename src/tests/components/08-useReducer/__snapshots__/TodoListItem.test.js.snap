// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem /> Debe de mostrarse correctamente 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    2
    . 
    Aprender Mongo
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
